pipenv/cli.py

```diff
@@ -398,14 +398,16 @@ def ensure_project(three=None, python=None, validate=True, system=False, warn=Tr
 
                 path_to_python = which('python')
 
-                if project.required_python_version not in python_version(path_to_python):
+                full_python_version = python_version(path_to_python)
+                if full_python_version is None or \
+                    project.required_python_version not in full_python_version:
                     puts(
                         '{0}: Your Pipfile requires {1} {2}, '
                         'but you are using {3} ({4}).'.format(
                             crayons.red('Warning', bold=True),
                             crayons.white('python_version', bold=True),
                             crayons.blue(project.required_python_version),
-                            crayons.blue(python_version(path_to_python)),
+                            crayons.blue(full_python_version or "<unknown>"),
```

Comment:
this code path should never be hit, `which('python')` should *always* return — if it's not, there's a bug we need to fix.

Reply:
It was getting hit here: https://ci.appveyor.com/project/kennethreitz/pipenv/build/1.0.549#L114

Reply:
yeah, there's a bug on windows with it. trying to figure it out...

Reply:
i think that missing assert might be the key

Reply:
As the code is implemented `python_version()` _can_ return `None`, so I think that either `python_version()` should raise an exception rather than return `None` or it's callers should treat `None` as a valid response and handle it.

Reply:
`python_version` will only return `None` of the executable it was passed isn't a Python, though..

Which I suppose, is possible, but is unlikely.